<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Pathway Analysis using ClusterProfile-R package | Portfolio DataScience of Khadija Zbair</title>
  <meta name="description" content="2 Pathway Analysis using ClusterProfile-R package | Portfolio DataScience of Khadija Zbair" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Pathway Analysis using ClusterProfile-R package | Portfolio DataScience of Khadija Zbair" />
  <meta property="og:type" content="book" />




  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Pathway Analysis using ClusterProfile-R package | Portfolio DataScience of Khadija Zbair" />




<meta name="author" content="Khadija Zbair" />


<meta name="date" content="2024-05-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />


<link rel="prev" href="future-outlook.html"/>
<link rel="next" href="guerrilla-analytics.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>
<script src="libs/core-js-2.5.3/shim.min.js"></script>
<script src="libs/react-18.2.0/react.min.js"></script>
<script src="libs/react-18.2.0/react-dom.min.js"></script>
<script src="libs/reactwidget-1.0.0/react-tools.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/reactable-0.4.4/reactable.css" rel="stylesheet" />
<script src="libs/reactable-binding-0.4.4/reactable.js"></script>
<script src="libs/plotly-binding-4.10.4/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.11.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">

  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><h3 class=".paddingtitel">Portfolio of Khadija Zbair</h3></li>
<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="curriculum-vitae.html"><a href="curriculum-vitae.html"><i class="fa fa-check"></i>Curriculum vitae</a>
<ul>
<li class="chapter" data-level="" data-path="curriculum-vitae.html"><a href="curriculum-vitae.html#profile"><i class="fa fa-check"></i>Profile</a></li>
<li class="chapter" data-level="" data-path="curriculum-vitae.html"><a href="curriculum-vitae.html#education"><i class="fa fa-check"></i>Education</a></li>
<li class="chapter" data-level="" data-path="curriculum-vitae.html"><a href="curriculum-vitae.html#relevant-courses-overview"><i class="fa fa-check"></i>Relevant courses overview</a></li>
<li class="chapter" data-level="" data-path="curriculum-vitae.html"><a href="curriculum-vitae.html#skills"><i class="fa fa-check"></i>Skills</a></li>
<li class="chapter" data-level="" data-path="curriculum-vitae.html"><a href="curriculum-vitae.html#job-experience"><i class="fa fa-check"></i>Job experience</a></li>
<li class="chapter" data-level="" data-path="curriculum-vitae.html"><a href="curriculum-vitae.html#hobbies"><i class="fa fa-check"></i>Hobbies</a></li>
<li class="chapter" data-level="" data-path="curriculum-vitae.html"><a href="curriculum-vitae.html#contact"><i class="fa fa-check"></i>Contact</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="future-outlook.html"><a href="future-outlook.html"><i class="fa fa-check"></i><b>1</b> Future outlook</a></li>
<li class="chapter" data-level="2" data-path="pathway-analysis-using-clusterprofile-r-package.html"><a href="pathway-analysis-using-clusterprofile-r-package.html"><i class="fa fa-check"></i><b>2</b> Pathway Analysis using ClusterProfile-R package</a>
<ul>
<li class="chapter" data-level="2.1" data-path="pathway-analysis-using-clusterprofile-r-package.html"><a href="pathway-analysis-using-clusterprofile-r-package.html#my-workflow"><i class="fa fa-check"></i><b>2.1</b> My workflow</a></li>
<li class="chapter" data-level="2.2" data-path="pathway-analysis-using-clusterprofile-r-package.html"><a href="pathway-analysis-using-clusterprofile-r-package.html#aim-of-this-project"><i class="fa fa-check"></i><b>2.2</b> Aim of this project</a></li>
<li class="chapter" data-level="2.3" data-path="pathway-analysis-using-clusterprofile-r-package.html"><a href="pathway-analysis-using-clusterprofile-r-package.html#clusterprofiler-package"><i class="fa fa-check"></i><b>2.3</b> Clusterprofiler package</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="pathway-analysis-using-clusterprofile-r-package.html"><a href="pathway-analysis-using-clusterprofile-r-package.html#installation"><i class="fa fa-check"></i><b>2.3.1</b> Installation</a></li>
<li class="chapter" data-level="2.3.2" data-path="pathway-analysis-using-clusterprofile-r-package.html"><a href="pathway-analysis-using-clusterprofile-r-package.html#gathering-data-and-annotations"><i class="fa fa-check"></i><b>2.3.2</b> Gathering data and annotations</a></li>
<li class="chapter" data-level="2.3.3" data-path="pathway-analysis-using-clusterprofile-r-package.html"><a href="pathway-analysis-using-clusterprofile-r-package.html#prepare-input"><i class="fa fa-check"></i><b>2.3.3</b> Prepare Input</a></li>
<li class="chapter" data-level="2.3.4" data-path="pathway-analysis-using-clusterprofile-r-package.html"><a href="pathway-analysis-using-clusterprofile-r-package.html#kegg-pathway-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>2.3.4</b> KEGG pathway gene set enrichment analysis</a></li>
<li class="chapter" data-level="2.3.5" data-path="pathway-analysis-using-clusterprofile-r-package.html"><a href="pathway-analysis-using-clusterprofile-r-package.html#kegg-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>2.3.5</b> KEGG Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="2.3.6" data-path="pathway-analysis-using-clusterprofile-r-package.html"><a href="pathway-analysis-using-clusterprofile-r-package.html#prepare-input-1"><i class="fa fa-check"></i><b>2.3.6</b> Prepare Input</a></li>
<li class="chapter" data-level="2.3.7" data-path="pathway-analysis-using-clusterprofile-r-package.html"><a href="pathway-analysis-using-clusterprofile-r-package.html#create-gsekegg-object"><i class="fa fa-check"></i><b>2.3.7</b> Create gseKEGG Object</a></li>
<li class="chapter" data-level="2.3.8" data-path="pathway-analysis-using-clusterprofile-r-package.html"><a href="pathway-analysis-using-clusterprofile-r-package.html#dotplot"><i class="fa fa-check"></i><b>2.3.8</b> Dotplot()</a></li>
<li class="chapter" data-level="2.3.9" data-path="pathway-analysis-using-clusterprofile-r-package.html"><a href="pathway-analysis-using-clusterprofile-r-package.html#pathview"><i class="fa fa-check"></i><b>2.3.9</b> Pathview</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="guerrilla-analytics.html"><a href="guerrilla-analytics.html"><i class="fa fa-check"></i><b>3</b> Guerrilla Analytics</a></li>
<li class="chapter" data-level="4" data-path="c.elegans-plate-experiment.html"><a href="c.elegans-plate-experiment.html"><i class="fa fa-check"></i><b>4</b> C.elegans plate experiment</a>
<ul>
<li class="chapter" data-level="4.1" data-path="c.elegans-plate-experiment.html"><a href="c.elegans-plate-experiment.html#importing-and-inspecting-the-data"><i class="fa fa-check"></i><b>4.1</b> Importing and inspecting the data</a></li>
<li class="chapter" data-level="4.2" data-path="c.elegans-plate-experiment.html"><a href="c.elegans-plate-experiment.html#checking-and-correcting-the-datatypes"><i class="fa fa-check"></i><b>4.2</b> Checking and correcting the datatypes</a></li>
<li class="chapter" data-level="4.3" data-path="c.elegans-plate-experiment.html"><a href="c.elegans-plate-experiment.html#scatterplot-with-log10-and-jitter"><i class="fa fa-check"></i><b>4.3</b> Scatterplot with log10 and jitter</a></li>
<li class="chapter" data-level="4.4" data-path="c.elegans-plate-experiment.html"><a href="c.elegans-plate-experiment.html#statistical-analysis"><i class="fa fa-check"></i><b>4.4</b> Statistical analysis</a></li>
<li class="chapter" data-level="4.5" data-path="c.elegans-plate-experiment.html"><a href="c.elegans-plate-experiment.html#normalizing-the-controlnegative-data-and-plotting"><i class="fa fa-check"></i><b>4.5</b> Normalizing the ‘controlNegative’ data and plotting</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="reproducible-research-peer-review.html"><a href="reproducible-research-peer-review.html"><i class="fa fa-check"></i><b>5</b> Reproducible Research: Peer review</a>
<ul>
<li class="chapter" data-level="5.1" data-path="reproducible-research-peer-review.html"><a href="reproducible-research-peer-review.html#choosing-the-right-article"><i class="fa fa-check"></i><b>5.1</b> Choosing the right article</a></li>
<li class="chapter" data-level="5.2" data-path="reproducible-research-peer-review.html"><a href="reproducible-research-peer-review.html#reference-to-the-article"><i class="fa fa-check"></i><b>5.2</b> Reference to the article</a></li>
<li class="chapter" data-level="5.3" data-path="reproducible-research-peer-review.html"><a href="reproducible-research-peer-review.html#transparicy-criteria"><i class="fa fa-check"></i><b>5.3</b> Transparicy criteria</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="reproducible-research-peer-review.html"><a href="reproducible-research-peer-review.html#general-aim"><i class="fa fa-check"></i><b>5.3.1</b> General aim</a></li>
<li class="chapter" data-level="5.3.2" data-path="reproducible-research-peer-review.html"><a href="reproducible-research-peer-review.html#methodology"><i class="fa fa-check"></i><b>5.3.2</b> Methodology</a></li>
<li class="chapter" data-level="5.3.3" data-path="reproducible-research-peer-review.html"><a href="reproducible-research-peer-review.html#results"><i class="fa fa-check"></i><b>5.3.3</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="reproducible-research-peer-review.html"><a href="reproducible-research-peer-review.html#conlusion"><i class="fa fa-check"></i><b>5.4</b> Conlusion</a></li>
<li class="chapter" data-level="5.5" data-path="reproducible-research-peer-review.html"><a href="reproducible-research-peer-review.html#reproducing-the-visuals-of-a-paper"><i class="fa fa-check"></i><b>5.5</b> Reproducing the visuals of a paper</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="reproducible-research-peer-review.html"><a href="reproducible-research-peer-review.html#the-code"><i class="fa fa-check"></i><b>5.5.1</b> The code</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="r-package.html"><a href="r-package.html"><i class="fa fa-check"></i><b>6</b> R-package</a>
<ul>
<li class="chapter" data-level="6.1" data-path="r-package.html"><a href="r-package.html#the-demo"><i class="fa fa-check"></i><b>6.1</b> The Demo</a></li>
<li class="chapter" data-level="6.2" data-path="r-package.html"><a href="r-package.html#r-markdown-driven-development"><i class="fa fa-check"></i><b>6.2</b> R-markdown driven development</a></li>
<li class="chapter" data-level="6.3" data-path="r-package.html"><a href="r-package.html#package-hanaahrtidy"><i class="fa fa-check"></i><b>6.3</b> Package “HanaahRTidy”</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="r-package.html"><a href="r-package.html#choosing-a-name"><i class="fa fa-check"></i><b>6.3.1</b> Choosing a name</a></li>
<li class="chapter" data-level="6.3.2" data-path="r-package.html"><a href="r-package.html#overview-package"><i class="fa fa-check"></i><b>6.3.2</b> Overview package</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="r-package.html"><a href="r-package.html#functionality"><i class="fa fa-check"></i><b>6.4</b> Functionality</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="antimicrobial-resistance-projecticum.html"><a href="antimicrobial-resistance-projecticum.html"><i class="fa fa-check"></i><b>7</b> Antimicrobial resistance projecticum</a>
<ul>
<li class="chapter" data-level="7.1" data-path="antimicrobial-resistance-projecticum.html"><a href="antimicrobial-resistance-projecticum.html#introduction-1"><i class="fa fa-check"></i><b>7.1</b> Introduction</a></li>
<li class="chapter" data-level="7.2" data-path="antimicrobial-resistance-projecticum.html"><a href="antimicrobial-resistance-projecticum.html#about-antimicrobial-resistance"><i class="fa fa-check"></i><b>7.2</b> About antimicrobial resistance</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="antimicrobial-resistance-projecticum.html"><a href="antimicrobial-resistance-projecticum.html#r-shiny-app"><i class="fa fa-check"></i><b>7.2.1</b> R shiny app</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="antimicrobial-resistance-projecticum.html"><a href="antimicrobial-resistance-projecticum.html#references"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="parameterized-covid-19-report.html"><a href="parameterized-covid-19-report.html"><i class="fa fa-check"></i><b>8</b> Parameterized COVID-19 report</a>
<ul>
<li class="chapter" data-level="8.1" data-path="parameterized-covid-19-report.html"><a href="parameterized-covid-19-report.html#inspecting-the-data"><i class="fa fa-check"></i><b>8.1</b> Inspecting the data</a></li>
<li class="chapter" data-level="8.2" data-path="parameterized-covid-19-report.html"><a href="parameterized-covid-19-report.html#visualising-the-data"><i class="fa fa-check"></i><b>8.2</b> Visualising the data</a></li>
<li class="chapter" data-level="8.3" data-path="parameterized-covid-19-report.html"><a href="parameterized-covid-19-report.html#parameters"><i class="fa fa-check"></i><b>8.3</b> Parameters</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references-1.html"><a href="references-1.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Portfolio DataScience of Khadija Zbair</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="pathway-analysis-using-clusterprofile-r-package" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">2</span> Pathway Analysis using ClusterProfile-R package<a href="pathway-analysis-using-clusterprofile-r-package.html#pathway-analysis-using-clusterprofile-r-package" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>For my independent project, I am dedicating my time to learning a new skill: conducting pathway analysis. Understanding the molecular mechanisms underlying diseases is paramount for developing effective treatments.The plan is to experiment with KEGG pathway analysis to become proficient in utilizing Clusterprofile, the R-package through which pathway analysis will be conducted. Initially, I will apply KEGG pathway analysis to Drosophila melanogaster data.</p>
<hr />
<div id="my-workflow" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> My workflow<a href="pathway-analysis-using-clusterprofile-r-package.html#my-workflow" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Day
</th>
<th style="text-align:left;">
Activities
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Day 1
</td>
<td style="text-align:left;">
<ol>
<li>
<b>Dataset Collection:</b> Figure out how the Clusterprofiler works and what input (format) it requires. Gather gene expression data. Maybe use data from the KEGG database.
</li>
<li>
<b>Data Preprocessing:</b> Import and preprocess the datasets in R, including gene normalization and filtering.
</li>
</ol>
</td>
</tr>
<tr>
<td style="text-align:left;">
Day 2
</td>
<td style="text-align:left;">
<ol>
<li>
<b>Differential Gene Expression Analysis:</b> Identify DEGs between lung cancer patients and healthy controls, specifically for female non-smokers. Do the same for the KEGG data.
</li>
<li>
<b>Filtering Significant Genes:</b> Select genes with statistically significant differential expression.
</li>
<li>
<b>Visual Inspection:</b> Create visualizations such as volcano plots and heatmaps for inspecting DEGs.
</li>
</ol>
</td>
</tr>
<tr>
<td style="text-align:left;">
Day 3
</td>
<td style="text-align:left;">
<ol>
<li>
<b>GO Term Enrichment Analysis:</b> Conduct GO term enrichment analysis to identify biological processes associated with differentially expressed genes.
</li>
<li>
<b>Interpretation:</b> Analyze and interpret the biological significance of the identified GO terms.
</li>
</ol>
</td>
</tr>
<tr>
<td style="text-align:left;">
Day 4
</td>
<td style="text-align:left;">
<ol>
<li>
<b>KEGG Pathway Analysis:</b> Perform KEGG pathway analysis using R packages such as clusterProfiler to map the identified DEGs to relevant biological pathways.
</li>
<li>
<b>Identify Key Pathways:</b> Select pathways significantly enriched with DEGs associated with lung cancer in female non-smokers.
</li>
<li>
<b>Summarize Results:</b> Write a summary of the identified pathways. Visualize them with <code>pathview()</code>.
</li>
</ol>
</td>
</tr>
</tbody>
</table>
</div>
<div id="aim-of-this-project" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Aim of this project<a href="pathway-analysis-using-clusterprofile-r-package.html#aim-of-this-project" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The aim is to acquire a new skill: conducting pathway analysis. For this, I need to master the Clusterprofiler package.</p>
<p>Learning <code>clusterProfiler</code> automates gene expression analysis with KEGG pathway enrichment, gene set enrichment, and over-representation analysis. Mastering it helps interpret complex datasets, crucial for understanding diseases like cancer.</p>
</div>
<div id="clusterprofiler-package" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Clusterprofiler package<a href="pathway-analysis-using-clusterprofile-r-package.html#clusterprofiler-package" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <code>clusterProfiler</code> package is a powerful tool for biologists and bioinformaticians to analyze and interpret gene expression data based on the KEGG database. It provides functionalities such as KEGG enrichment analysis to determine which biological pathways may be involved in a phenotype or condition. Furthermore, it supports gene set enrichment analysis for specific KEGG pathway terms, allowing researchers to assess whether genes involved in certain biological functions are enriched in their dataset.</p>
<p>Another valuable functionality is the over-representation analysis of KEGG modules, which provides a more structured view of biological processes. The package also offers various visualization methods, such as the <code>browseKEGG</code> function to explore enriched KEGG pathway terms in a web browser and the <code>pathview</code> function to present them visually, as shown in Figure 3.1.</p>
<p>In summary, <code>clusterProfiler</code> is a valuable tool for interpreting gene expression data in the context of biological processes and pathways, enabling researchers to gain new insights into complex biological mechanisms.</p>
<p>More information on Clusterprofiler can be found here: <a href="https://yulab-smu.top/biomedical-knowledge-mining-book/clusterprofiler-kegg.html" class="uri">https://yulab-smu.top/biomedical-knowledge-mining-book/clusterprofiler-kegg.html</a>.</p>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-5"></span>
<img src="Images/pathview.png" alt="Visualize the selected KEGG pathway using pathview(). Gene expression values can be assigned to a gradient color scale." width="70%" />
<p class="caption">
Figure 2.1: Visualize the selected KEGG pathway using pathview(). Gene expression values can be assigned to a gradient color scale.
</p>
</div>
<p>For performing a pathway analysis with ClusterProfiler, we go through several steps. First, we load the required R packages, including ClusterProfiler and the corresponding annotation database. Next, we prepare our input data by reading and processing it, such as selecting relevant data and removing missing values. Then, we conduct the actual analysis, utilizing functions like gseGO for gene ontology (GO) terms and gseKEGG for KEGG pathway analysis. These steps assist us in gaining insights into which biological pathways may be affected by changes in gene expression.</p>
<div id="installation" class="section level3 hasAnchor" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> Installation<a href="pathway-analysis-using-clusterprofile-r-package.html#installation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The installation of the Clusterprofiler package is described <a href="https://bioconductor.org/packages/release/bioc/html/clusterProfiler.html">here</a>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb1-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-2"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb1-2" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb1-3"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb1-3" tabindex="-1"></a></span>
<span id="cb1-4"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb1-4" tabindex="-1"></a><span class="co"># Loading the required libraries</span></span>
<span id="cb1-5"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb1-5" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb1-6"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb1-6" tabindex="-1"></a><span class="fu">library</span>(airway)</span>
<span id="cb1-7"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb1-7" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb1-8"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb1-8" tabindex="-1"></a><span class="fu">library</span>(org.Dm.eg.db)</span>
<span id="cb1-9"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb1-9" tabindex="-1"></a><span class="fu">library</span>(pathview)</span>
<span id="cb1-10"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb1-10" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-11"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb1-11" tabindex="-1"></a><span class="fu">library</span>(enrichplot)</span></code></pre></div>
</div>
<div id="gathering-data-and-annotations" class="section level3 hasAnchor" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> Gathering data and annotations<a href="pathway-analysis-using-clusterprofile-r-package.html#gathering-data-and-annotations" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>I’m analyzing Drosophila melanogaster data, so I install and load the annotation package “org.Dm.eg.db” <a href="http://bioconductor.org/packages/release/BiocViews.html#___OrgDb">here</a>. There are also other available annotations .</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb2-1" tabindex="-1"></a><span class="co"># Setting the organism (I chose fly for example ) </span></span>
<span id="cb2-2"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb2-2" tabindex="-1"></a>organism <span class="ot">=</span> <span class="st">&quot;org.Dm.eg.db&quot;</span></span>
<span id="cb2-3"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb2-3" tabindex="-1"></a></span>
<span id="cb2-4"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb2-4" tabindex="-1"></a><span class="co">#BiocManager::install(organism, character.only = TRUE)</span></span>
<span id="cb2-5"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb2-5" tabindex="-1"></a><span class="fu">library</span>(organism, <span class="at">character.only =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>Before diving into the analysis, we first need to prepare our input data. In this step, we’re reading in data obtained from differential expression analysis using DESeq2. We load the dataset named “drosphila_example_de.csv”, which includes important information such as log2 fold change values.</p>
</div>
<div id="prepare-input" class="section level3 hasAnchor" number="2.3.3">
<h3><span class="header-section-number">2.3.3</span> Prepare Input<a href="pathway-analysis-using-clusterprofile-r-package.html#prepare-input" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb3-1" tabindex="-1"></a><span class="co"># Reading in data from deseq2</span></span>
<span id="cb3-2"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb3-2" tabindex="-1"></a>drosphilia_data <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;drosphila_example_de.csv&quot;</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb3-3"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb3-3" tabindex="-1"></a></span>
<span id="cb3-4"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb3-4" tabindex="-1"></a><span class="co"># I want the log2 fold change  </span></span>
<span id="cb3-5"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb3-5" tabindex="-1"></a>original_gene_list <span class="ot">&lt;-</span> drosphilia_data<span class="sc">$</span>log2FoldChange</span>
<span id="cb3-6"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb3-6" tabindex="-1"></a></span>
<span id="cb3-7"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb3-7" tabindex="-1"></a><span class="co"># Naming the vector</span></span>
<span id="cb3-8"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb3-8" tabindex="-1"></a><span class="fu">names</span>(original_gene_list) <span class="ot">&lt;-</span> drosphilia_data<span class="sc">$</span>X</span>
<span id="cb3-9"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb3-9" tabindex="-1"></a></span>
<span id="cb3-10"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb3-10" tabindex="-1"></a><span class="co"># Remove the missing values</span></span>
<span id="cb3-11"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb3-11" tabindex="-1"></a>gene_list<span class="ot">&lt;-</span><span class="fu">na.omit</span>(original_gene_list)</span>
<span id="cb3-12"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb3-12" tabindex="-1"></a></span>
<span id="cb3-13"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb3-13" tabindex="-1"></a><span class="co"># Sort the list in decreasing order (a must for clusterProfiler)</span></span>
<span id="cb3-14"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb3-14" tabindex="-1"></a>gene_list <span class="ot">=</span> <span class="fu">sort</span>(gene_list, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>After preparing our input data, we proceed with gene set enrichment analysis to unravel biological insights from our dataset. Using the gseGO function, we identify enriched gene ontology (GO) terms associated with the differentially expressed genes.</p>
</div>
<div id="kegg-pathway-gene-set-enrichment-analysis" class="section level3 hasAnchor" number="2.3.4">
<h3><span class="header-section-number">2.3.4</span> KEGG pathway gene set enrichment analysis<a href="pathway-analysis-using-clusterprofile-r-package.html#kegg-pathway-gene-set-enrichment-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb4-1" tabindex="-1"></a>gene_set_enrichment <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList=</span>gene_list, </span>
<span id="cb4-2"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb4-2" tabindex="-1"></a>             <span class="at">ont =</span><span class="st">&quot;ALL&quot;</span>, </span>
<span id="cb4-3"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb4-3" tabindex="-1"></a>             <span class="at">keyType =</span> <span class="st">&quot;ENSEMBL&quot;</span>, </span>
<span id="cb4-4"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb4-4" tabindex="-1"></a>             <span class="at">nPerm =</span> <span class="dv">10000</span>, </span>
<span id="cb4-5"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb4-5" tabindex="-1"></a>             <span class="at">minGSSize =</span> <span class="dv">3</span>, </span>
<span id="cb4-6"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb4-6" tabindex="-1"></a>             <span class="at">maxGSSize =</span> <span class="dv">800</span>, </span>
<span id="cb4-7"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb4-7" tabindex="-1"></a>             <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>, </span>
<span id="cb4-8"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb4-8" tabindex="-1"></a>             <span class="at">verbose =</span> <span class="cn">TRUE</span>, </span>
<span id="cb4-9"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb4-9" tabindex="-1"></a>             <span class="at">OrgDb =</span> organism, </span>
<span id="cb4-10"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb4-10" tabindex="-1"></a>             <span class="at">pAdjustMethod =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb4-11"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb4-11" tabindex="-1"></a></span>
<span id="cb4-12"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb4-12" tabindex="-1"></a><span class="co"># The results </span></span>
<span id="cb4-13"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb4-13" tabindex="-1"></a><span class="fu">head</span>(gene_set_enrichment)</span></code></pre></div>
<pre><code>##            ONTOLOGY         ID
## GO:0007186       BP GO:0007186
## GO:0004930       MF GO:0004930
## GO:0022626       CC GO:0022626
## GO:0016798       MF GO:0016798
## GO:0007188       BP GO:0007188
## GO:0007218       BP GO:0007218
##                                                                          Description
## GO:0007186                              G protein-coupled receptor signaling pathway
## GO:0004930                                       G protein-coupled receptor activity
## GO:0022626                                                        cytosolic ribosome
## GO:0016798                              hydrolase activity, acting on glycosyl bonds
## GO:0007188 adenylate cyclase-modulating G protein-coupled receptor signaling pathway
## GO:0007218                                            neuropeptide signaling pathway
##            setSize enrichmentScore       NES       pvalue     p.adjust   qvalue
## GO:0007186     227      -0.5558057 -2.117391 0.0001423690 0.0001423690 0.103108
## GO:0004930     113      -0.5809564 -2.025400 0.0001530456 0.0001530456 0.103108
## GO:0022626     106      -0.5200908 -1.796688 0.0001540832 0.0001540832 0.103108
## GO:0016798      95      -0.5240824 -1.784996 0.0001549667 0.0001549667 0.103108
## GO:0007188      63      -0.6926563 -2.207667 0.0001639076 0.0001639076 0.103108
## GO:0007218      55      -0.6056430 -1.886525 0.0001661130 0.0001661130 0.103108
##            rank                   leading_edge
## GO:0007186  885  tags=25%, list=6%, signal=24%
## GO:0004930 1016  tags=32%, list=7%, signal=30%
## GO:0022626 5512 tags=71%, list=36%, signal=45%
## GO:0016798 2794 tags=34%, list=18%, signal=28%
## GO:0007188 1344  tags=41%, list=9%, signal=38%
## GO:0007218  569  tags=35%, list=4%, signal=33%
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                core_enrichment
## GO:0007186                                                                                                                                                                                                                                     FBgn0050361/FBgn0038880/FBgn0040506/FBgn0001104/FBgn0050340/FBgn0260446/FBgn0022710/FBgn0000253/FBgn0053639/FBgn0031275/FBgn0004842/FBgn0004573/FBgn0004622/FBgn0052547/FBgn0033932/FBgn0027794/FBgn0038140/FBgn0050106/FBgn0024944/FBgn0052683/FBgn0266137/FBgn0263131/FBgn0028956/FBgn0010223/FBgn0038653/FBgn0011582/FBgn0038542/FBgn0086704/FBgn0085380/FBgn0004784/FBgn0265959/FBgn0030437/FBgn0267252/FBgn0261953/FBgn0013767/FBgn0039396/FBgn0038980/FBgn0032048/FBgn0037976/FBgn0031302/FBgn0264002/FBgn0031770/FBgn0037408/FBgn0035092/FBgn0036260/FBgn0033744/FBgn0035385/FBgn0266429/FBgn0085413/FBgn0260753/FBgn0019985/FBgn0045038/FBgn0261549/FBgn0034636/FBgn0039419/FBgn0052843
## GO:0004930                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     FBgn0263116/FBgn0035331/FBgn0050361/FBgn0038880/FBgn0050340/FBgn0260446/FBgn0053639/FBgn0031275/FBgn0004842/FBgn0004573/FBgn0004622/FBgn0052547/FBgn0033932/FBgn0038140/FBgn0050106/FBgn0024944/FBgn0266137/FBgn0028956/FBgn0038653/FBgn0011582/FBgn0038542/FBgn0025680/FBgn0030437/FBgn0039396/FBgn0038980/FBgn0264002/FBgn0031770/FBgn0037408/FBgn0036260/FBgn0033744/FBgn0035385/FBgn0266429/FBgn0260753/FBgn0019985/FBgn0039419/FBgn0052843
## GO:0022626 FBgn0053368/FBgn0053403/FBgn0053411/FBgn0053412/FBgn0053420/FBgn0053421/FBgn0053425/FBgn0053426/FBgn0011284/FBgn0002626/FBgn0032987/FBgn0010265/FBgn0261592/FBgn0016726/FBgn0066084/FBgn0030616/FBgn0033555/FBgn0003274/FBgn0037351/FBgn0034968/FBgn0017545/FBgn0029897/FBgn0064225/FBgn0039757/FBgn0019936/FBgn0010198/FBgn0010078/FBgn0011272/FBgn0004403/FBgn0002593/FBgn0039713/FBgn0261599/FBgn0020618/FBgn0002590/FBgn0003942/FBgn0034837/FBgn0261602/FBgn0036825/FBgn0037135/FBgn0010411/FBgn0005593/FBgn0035753/FBgn0261596/FBgn0039406/FBgn0030483/FBgn0038834/FBgn0086710/FBgn0014026/FBgn0034743/FBgn0029148/FBgn0023170/FBgn0030136/FBgn0002579/FBgn0003517/FBgn0034138/FBgn0032518/FBgn0035422/FBgn0029785/FBgn0004404/FBgn0004867/FBgn0053452/FBgn0036213/FBgn0034822/FBgn0037899/FBgn0037686/FBgn0039857/FBgn0003279/FBgn0026372/FBgn0261608/FBgn0010408/FBgn0261597/FBgn0015521/FBgn0039359/FBgn0038277/FBgn0015288
## GO:0016798                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     FBgn0032067/FBgn0034538/FBgn0034165/FBgn0053080/FBgn0004428/FBgn0002570/FBgn0027588/FBgn0032066/FBgn0039634/FBgn0050360/FBgn0033294/FBgn0027611/FBgn0053054/FBgn0032381/FBgn0026869/FBgn0032907/FBgn0029913/FBgn0259170/FBgn0030510/FBgn0023511/FBgn0030839/FBgn0026616/FBgn0031265/FBgn0034117/FBgn0041630/FBgn0032382/FBgn0032069/FBgn0030511/FBgn0262579/FBgn0263132/FBgn0260746/FBgn0028499
## GO:0007188                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             FBgn0040507/FBgn0050054/FBgn0000037/FBgn0037546/FBgn0263116/FBgn0035331/FBgn0040506/FBgn0001104/FBgn0260446/FBgn0022710/FBgn0004573/FBgn0033932/FBgn0266137/FBgn0263131/FBgn0010223/FBgn0038653/FBgn0011582/FBgn0038542/FBgn0030437/FBgn0038980/FBgn0032048/FBgn0264002/FBgn0033744/FBgn0260753/FBgn0039419/FBgn0052843
## GO:0007218                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 FBgn0011581/FBgn0035331/FBgn0038880/FBgn0004842/FBgn0004622/FBgn0052547/FBgn0038140/FBgn0050106/FBgn0013767/FBgn0039396/FBgn0037976/FBgn0264002/FBgn0031770/FBgn0037408/FBgn0035092/FBgn0035385/FBgn0266429/FBgn0260753/FBgn0045038</code></pre>
</div>
<div id="kegg-gene-set-enrichment-analysis" class="section level3 hasAnchor" number="2.3.5">
<h3><span class="header-section-number">2.3.5</span> KEGG Gene Set Enrichment Analysis<a href="pathway-analysis-using-clusterprofile-r-package.html#kegg-gene-set-enrichment-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To find KEGG pathways related to our genes using <code>gseKEGG()</code>, we first need to change how our gene identifiers are labeled. We do this with the <code>bitr</code> function in clusterProfiler.s After this, we use our original gene list to start the analysis.</p>
</div>
<div id="prepare-input-1" class="section level3 hasAnchor" number="2.3.6">
<h3><span class="header-section-number">2.3.6</span> Prepare Input<a href="pathway-analysis-using-clusterprofile-r-package.html#prepare-input-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-1" tabindex="-1"></a><span class="co">#Convert gene IDs for gseKEGG function</span></span>
<span id="cb6-2"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-2" tabindex="-1"></a>ids<span class="ot">&lt;-</span><span class="fu">bitr</span>(<span class="fu">names</span>(original_gene_list), <span class="at">fromType =</span> <span class="st">&quot;ENSEMBL&quot;</span>, <span class="at">toType =</span> <span class="st">&quot;ENTREZID&quot;</span>, <span class="at">OrgDb=</span>organism)</span>
<span id="cb6-3"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-3" tabindex="-1"></a></span>
<span id="cb6-4"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-4" tabindex="-1"></a><span class="co"># remove duplicate IDS (here I use &quot;ENSEMBL&quot;, but it should be whatever was selected as keyType)</span></span>
<span id="cb6-5"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-5" tabindex="-1"></a>dedup_ids <span class="ot">=</span> ids[<span class="sc">!</span><span class="fu">duplicated</span>(ids[<span class="fu">c</span>(<span class="st">&quot;ENSEMBL&quot;</span>)]),]</span>
<span id="cb6-6"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-6" tabindex="-1"></a></span>
<span id="cb6-7"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-7" tabindex="-1"></a><span class="co"># Create a new dataframe df2 which has only the genes which were successfully mapped using the bitr function above</span></span>
<span id="cb6-8"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-8" tabindex="-1"></a>df2 <span class="ot">=</span> df[df<span class="sc">$</span>X <span class="sc">%in%</span> dedup_ids<span class="sc">$</span>ENSEMBL,]</span>
<span id="cb6-9"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-9" tabindex="-1"></a></span>
<span id="cb6-10"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-10" tabindex="-1"></a><span class="co"># Create a new column in df2 with the corresponding ENTREZ IDs</span></span>
<span id="cb6-11"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-11" tabindex="-1"></a>df2<span class="sc">$</span>Y <span class="ot">=</span> dedup_ids<span class="sc">$</span>ENTREZID</span>
<span id="cb6-12"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-12" tabindex="-1"></a></span>
<span id="cb6-13"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-13" tabindex="-1"></a><span class="co"># Create a vector of the gene unuiverse</span></span>
<span id="cb6-14"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-14" tabindex="-1"></a>kegg_gene_list <span class="ot">&lt;-</span> df2<span class="sc">$</span>log2FoldChange</span>
<span id="cb6-15"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-15" tabindex="-1"></a></span>
<span id="cb6-16"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-16" tabindex="-1"></a><span class="co"># Name vector with ENTREZ ids</span></span>
<span id="cb6-17"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-17" tabindex="-1"></a><span class="fu">names</span>(kegg_gene_list) <span class="ot">&lt;-</span> df2<span class="sc">$</span>Y</span>
<span id="cb6-18"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-18" tabindex="-1"></a></span>
<span id="cb6-19"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-19" tabindex="-1"></a><span class="co"># omit any NA values </span></span>
<span id="cb6-20"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-20" tabindex="-1"></a>kegg_gene_list<span class="ot">&lt;-</span><span class="fu">na.omit</span>(kegg_gene_list)</span>
<span id="cb6-21"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-21" tabindex="-1"></a></span>
<span id="cb6-22"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-22" tabindex="-1"></a><span class="co"># sort the list in decreasing order (required for clusterProfiler)</span></span>
<span id="cb6-23"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb6-23" tabindex="-1"></a>kegg_gene_list <span class="ot">=</span> <span class="fu">sort</span>(kegg_gene_list, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="create-gsekegg-object" class="section level3 hasAnchor" number="2.3.7">
<h3><span class="header-section-number">2.3.7</span> Create gseKEGG Object<a href="pathway-analysis-using-clusterprofile-r-package.html#create-gsekegg-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To create the gseKEGG object for our analysis, we need to set several parameters:</p>
<ul>
<li><p>organism: This refers to the KEGG organism code, which consists of a three-letter code identifying the organism of interest. You can find the full list of KEGG organism codes here. We define this as kegg_organism initially because it will be used again later when generating pathview plots.</p></li>
<li><p>nPerm: This parameter determines the number of permutations used in the analysis. Increasing the number of permutations can improve the accuracy of results but will also increase computation time.</p></li>
<li><p>minGSSize: This sets the minimum size of each gene set for analysis.</p></li>
<li><p>maxGSSize: This sets the maximum size of genes annotated for testing.</p></li>
<li><p>pvalueCutoff: This parameter specifies the cutoff value for statistical significance.</p></li>
<li><p>pAdjustMethod: This determines the method used for adjusting p-values. Options include “holm”, “hochberg”, “hommel”, “bonferroni”, “BH”, “BY”, “fdr”, and “none”.</p></li>
<li><p>keyType: This parameter specifies the type of gene identifiers used in the analysis. Options include ‘kegg’, ‘ncbi-geneid’, ‘ncib-proteinid’, or ‘uniprot’.</p></li>
</ul>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb7-1" tabindex="-1"></a>kegg_organism <span class="ot">=</span> <span class="st">&quot;dme&quot;</span></span>
<span id="cb7-2"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb7-2" tabindex="-1"></a>kk2 <span class="ot">&lt;-</span> <span class="fu">gseKEGG</span>(<span class="at">geneList     =</span> kegg_gene_list,</span>
<span id="cb7-3"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb7-3" tabindex="-1"></a>               <span class="at">organism     =</span> kegg_organism,</span>
<span id="cb7-4"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb7-4" tabindex="-1"></a>               <span class="at">nPerm        =</span> <span class="dv">10000</span>,</span>
<span id="cb7-5"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb7-5" tabindex="-1"></a>               <span class="at">minGSSize    =</span> <span class="dv">3</span>,</span>
<span id="cb7-6"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb7-6" tabindex="-1"></a>               <span class="at">maxGSSize    =</span> <span class="dv">800</span>,</span>
<span id="cb7-7"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb7-7" tabindex="-1"></a>               <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb7-8"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb7-8" tabindex="-1"></a>               <span class="at">pAdjustMethod =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb7-9"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb7-9" tabindex="-1"></a>               <span class="at">keyType       =</span> <span class="st">&quot;ncbi-geneid&quot;</span>)</span>
<span id="cb7-10"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb7-10" tabindex="-1"></a><span class="fu">head</span>(kk2, <span class="dv">10</span>)</span></code></pre></div>
<pre><code>##                ID
## dme04080 dme04080
## dme00053 dme00053
## dme04310 dme04310
## dme00511 dme00511
## dme00330 dme00330
## dme04130 dme04130
## dme00071 dme00071
## dme00830 dme00830
## dme00380 dme00380
## dme00650 dme00650
##                                                                              Description
## dme04080   Neuroactive ligand-receptor interaction - Drosophila melanogaster (fruit fly)
## dme00053         Ascorbate and aldarate metabolism - Drosophila melanogaster (fruit fly)
## dme04310                     Wnt signaling pathway - Drosophila melanogaster (fruit fly)
## dme00511                  Other glycan degradation - Drosophila melanogaster (fruit fly)
## dme00330           Arginine and proline metabolism - Drosophila melanogaster (fruit fly)
## dme04130 SNARE interactions in vesicular transport - Drosophila melanogaster (fruit fly)
## dme00071                    Fatty acid degradation - Drosophila melanogaster (fruit fly)
## dme00830                        Retinol metabolism - Drosophila melanogaster (fruit fly)
## dme00380                     Tryptophan metabolism - Drosophila melanogaster (fruit fly)
## dme00650                      Butanoate metabolism - Drosophila melanogaster (fruit fly)
##          setSize enrichmentScore       NES       pvalue     p.adjust     qvalue
## dme04080      60      -0.6214491 -1.957696 0.0001641228 0.0001641228 0.02142234
## dme00053      32      -0.6646835 -1.852728 0.0013959169 0.0013959169 0.09110195
## dme04310      92       0.4194541  1.533753 0.0069003588 0.0069003588 0.20795483
## dme00511      21      -0.6797063 -1.728270 0.0073489873 0.0073489873 0.20795483
## dme00330      47      -0.5419275 -1.623696 0.0094626563 0.0094626563 0.20795483
## dme04130      20       0.6327935  1.696903 0.0123706703 0.0123706703 0.20795483
## dme00071      32      -0.5858576 -1.633010 0.0123887629 0.0123887629 0.20795483
## dme00830      31      -0.5867306 -1.626001 0.0139445703 0.0139445703 0.20795483
## dme00380      18      -0.6744193 -1.665382 0.0147349506 0.0147349506 0.20795483
## dme00650      15      -0.7008794 -1.663396 0.0166787527 0.0166787527 0.20795483
##          rank                   leading_edge
## dme04080 1364  tags=37%, list=9%, signal=33%
## dme00053   60  tags=16%, list=0%, signal=16%
## dme04310 2391 tags=27%, list=16%, signal=23%
## dme00511 1575 tags=57%, list=11%, signal=51%
## dme00330 2290 tags=26%, list=16%, signal=22%
## dme04130 2941 tags=50%, list=20%, signal=40%
## dme00071 2984 tags=44%, list=20%, signal=35%
## dme00830 2024 tags=26%, list=14%, signal=22%
## dme00380  290  tags=22%, list=2%, signal=22%
## dme00650 2770 tags=53%, list=19%, signal=43%
##                                                                                                                                                core_enrichment
## dme04080                 41017/39409/37892/40955/37191/33360/42530/34878/33248/43669/43551/36601/41639/37004/43484/41726/2768688/36368/44484/43838/38742/36475
## dme00053                                                                                                                         53584/53507/35138/34256/35139
## dme04310 39596/34011/45307/45343/32188/37713/31151/48311/42692/31310/34367/31597/40090/44317/34819/33204/43769/43319/31659/39884/34010/39751/32132/38890/31014
## dme00511                                                                               34204/37211/34203/34437/37490/32726/41913/38528/34206/35989/41457/42178
## dme00330                                                                             38986/326112/40026/46717/43620/34495/41167/40028/50001/43625/326188/34256
## dme04130                                                                                          36080/40094/42215/36015/248102/38614/40373/38541/40724/39051
## dme00071                                                                 37446/34276/34315/34313/3771877/41311/37445/41480/40059/42364/31695/37217/35213/34256
## dme00830                                                                                                     53513/3771877/41311/38598/53584/53507/35138/35139
## dme00380                                                                                                                               31695/43689/35190/34256
## dme00650                                                                                                       34276/38261/41480/42364/31695/30972/38484/35156</code></pre>
<p>After creating the gseKEGG object with specified parameters, we proceed to visualize the enriched pathways using a dotplot. This visualization method provides a clear overview of the enriched KEGG pathways identified in our analysis.</p>
<p>Each dot represents a pathway, and the position and size of the dots indicate the significance and number of genes involved in each pathway, respectively. The dotplot allows for easy comparison and interpretation of pathway enrichment results.</p>
</div>
<div id="dotplot" class="section level3 hasAnchor" number="2.3.8">
<h3><span class="header-section-number">2.3.8</span> Dotplot()<a href="pathway-analysis-using-clusterprofile-r-package.html#dotplot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb9-1" tabindex="-1"></a><span class="fu">dotplot</span>(kk2, <span class="at">showCategory =</span> <span class="dv">10</span>, <span class="at">title =</span> <span class="st">&quot;Enriched Pathways in the genes of </span></span>
<span id="cb9-2"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb9-2" tabindex="-1"></a><span class="st">        drosphila&quot;</span>, <span class="at">split=</span><span class="st">&quot;.sign&quot;</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>(.<span class="sc">~</span>.sign)</span></code></pre></div>
<p><img src="Images/_main_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
</div>
<div id="pathview" class="section level3 hasAnchor" number="2.3.9">
<h3><span class="header-section-number">2.3.9</span> Pathview<a href="pathway-analysis-using-clusterprofile-r-package.html#pathview" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Pathview() generates visual representations of enriched KEGG pathways, aiding in the interpretation of gene expression data.The following parameters are used:</p>
<ul>
<li><p>gene.data: This parameter corresponds to the kegg_gene_list created earlier in our analysis. It contains the gene expression data necessary for pathway visualization.</p></li>
<li><p>pathway.id: The user needs to input the pathway ID of interest. Information on enriched pathways, including their IDs, can be found in the output table of the gseKEGG function.</p></li>
<li><p>species: This parameter should be the same as the organism specified in the gseKEGG function, which we defined earlier as kegg_organism. It denotes the KEGG organism code for the species under investigation.</p></li>
</ul>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb10-1" tabindex="-1"></a><span class="fu">library</span>(pathview)</span>
<span id="cb10-2"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb10-2" tabindex="-1"></a></span>
<span id="cb10-3"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb10-3" tabindex="-1"></a><span class="co"># Produce the native KEGG plot</span></span>
<span id="cb10-4"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb10-4" tabindex="-1"></a>dme <span class="ot">&lt;-</span> <span class="fu">pathview</span>(<span class="at">gene.data=</span></span>
<span id="cb10-5"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb10-5" tabindex="-1"></a>                  kegg_gene_list, </span>
<span id="cb10-6"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb10-6" tabindex="-1"></a>                <span class="at">pathway.id=</span><span class="st">&quot;dme04130&quot;</span>, </span>
<span id="cb10-7"><a href="pathway-analysis-using-clusterprofile-r-package.html#cb10-7" tabindex="-1"></a>                <span class="at">species =</span> kegg_organism)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-14"></span>
<img src="Images/dme04130.pathview.png" alt="Pathways involved" width="70%" />
<p class="caption">
Figure 2.2: Pathways involved
</p>
</div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="future-outlook.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="guerrilla-analytics.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
